pyuvdata/uvdata/uvfits.py

```diff
@@ -134,16 +134,50 @@ def _get_parameter_data(
         # initialize internal variables based on the antenna lists
         self.Nants_data = int(np.union1d(self.ant_1_array, self.ant_2_array).size)
 
+        # check for suffixes in the baseline coordinate names indicating the
+        # baseline coordinate system
+        if (
+            "UU" in vis_hdu.data.parnames
+            and "VV" in vis_hdu.data.parnames
+            and "WW" in vis_hdu.data.parnames
+        ):
+            uvw_names = ["UU", "VV", "WW"]
+        elif (
+            "UU---SIN" in vis_hdu.data.parnames
+            and "VV---SIN" in vis_hdu.data.parnames
+            and "WW---SIN" in vis_hdu.data.parnames
+        ):
+            uvw_names = ["UU---SIN", "VV---SIN", "WW---SIN"]
+        elif (
+            "UU---NCP" in vis_hdu.data.parnames
+            and "VV---NCP" in vis_hdu.data.parnames
+            and "WW---NCP" in vis_hdu.data.parnames
+        ):
+            uvw_names = ["UU---NCP", "VV---NCP", "WW---NCP"]
+            warnings.warn(
+                "The baseline coordinates (uvws) in this file are specified in the "
+                "---NCP coordinate system, which is does not agree with our baseline "
+                "coordinate conventions. The values are not being modified to align "
+                "with our convention."
+            )
```

Comment:
I've been reading up on this a little bit, but am still confused. NCP _should_ just be an euler rotation (or 2? 3?) away from SIN projection right? Can we rotate the NCPs to the frame we expect using other metadata? @kartographer do you have any ideas? Or do I have the wrong idea about this frame in my head.

Reply:
@mkolopanis -- I believe you are correct, and we should be able to rotate the uvws to the right position with the source information alone (namely the apparent dec).

Reply:
Ok. Do you know what the rotation should be? I don't quite have it in my head.

Reply:
(To answer your question about why not rotate it @mkolopanis, I was being lazy on the assumption that it will rarely come up. But I think you're right, we should probably fix them up.)

Reply:
@bhazelton -- I think the line of code you need is just the following:
```
new_uvw = uvutils._rotate_one_axis(old_uvw[:, :, None], phase_center_app_dec - np.pi/2, 0)[:, :, 0]
```
This will "drop" the w-axis by rotating around the u-axis, which should give you the uvw's in the more typical form that we'd expect.

Reply:
Ok, I've added that and I have a test that just rotates back when checking for equality. Is there a better way to test that this rotation does exactly what we want?

I'm also not sure if what I'm doing properly handles cases where there are multiple phase centers. The `phase_center_app_dec` attribute is length `Nblts` so maybe it does?

Reply:
With respect to multiple phase centers, yes, that's basically handled by the fact that `phase_center_app_dec` is calculated on a per-baseline-time basis, so no worries there.

As for tests, the _only_ think I could think of would be maybe to construct a east-west interferometer/array and verify that for some given position in the sky, the NCP-projected coordinates actually have zero in the NCP-w term? I don't know how practical that is, but it's the only additional thing that comes to mind.

Reply:
As discussed on the telecon, I've added a warning back so the user will be aware that a rotation has happened.